### PR TITLE
DG: Added Acknowledgements and Implementation Details

### LINKED ISSUE #168: DG: add Create and Delete session

### LINKED ISSUE #169: DG: Add JsonAdaptedModuleClass and JsonAdaptedSession to Storage details

### LINKED ISSUE #171: DG: Update Acknowledgements

### PR DESCRIPTION
**Pull request by Berted**
Changes:
- Updated Acknowledgements section. Resolves #171.
- Added implementation details on immutability of certain classes
- Added implementation details on identity.
- Added implementation details on Session management. Resolves #168  (Albeit, partially, decided against writing about deleting sessions until issue #149 is resolved)
- Updated Storage to now show `JsonAdaptedSession`. Resolves #169.
- Updated PPP

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -167,53 +168,17 @@ Classes used by multiple components are in the `seedu.taassist.commons` package.
...
+for why certain functions are implemented in such a manner.
 
-- `TaAssist#addModuleClass(ModuleClass moduleClass)` - Adds the provided module class to the list of module classes created.
-- `TaAssist#removeModuleClass(ModuleClass moduleClass)` - Removes the provided module class from the list of module classes created.
-
-### Assigning students to module classes
+### Assigning students to a class
 <img src="images/AssignCommandSequenceDiagram.png" width="700" />
 
-Each student object contains a collection of `StudentModuleData` where module classes and the grades the student obtained for the sessions of the module classes are stored. When the user assigns students to a module class, a new `StudentModuleData` object is created and added to the collection for each student.
```
Comment (optionalemon) [inline docs/DeveloperGuide.md]: I think this was added by @Bubbl3T, and we probably need to keep these parts (i.e. Creating and deleting module classes and Assigning students to module classes so on and so for)

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -269,21 +289,21 @@ On the other hand, the `unfocus` command deactivates focus mode by setting `focu
...
-| `* * *`  | User            | Delete students from TA Assist           | Keep my list of students concise.        |
-| `* * *`  | User            | View all my classes                      | See what classes I am teaching.          |
-| `* *  `  | Infrequent user | Remember the last used commands          | Quickly find the commands that I need.   |
-| `* *  `  | An expert user  | Create macros to perform multiple tasks  | Be more efficient at using the system.   |
-| `* *  `  | User            | Edit students' information               | Easily and quickly update their information. |
-| `* * *`  | User            | Create assignments that contribute to CA components | Assign grades to assignments done by students. |
-| `* *  `  | User            | Assign weightage to my created assignments | Estimate the overall performance of my students. |
-| `* * *`  | User            | Change participation marks previously given to my students | Correctly and accurately reflect the marks for my students. |
+| Priority | As a …​         | I want to …​                                               | So that I can…​                                                  |
```
Comment (optionalemon) [inline docs/DeveloperGuide.md]: Similarly, I think we need to keep this part too

**Review by optionalemon — CHANGES_REQUESTED** [review]
LGTM! Maybe just slight changes to keep @Bubbl3T 's work?

**Review by Bubbl3T — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (Berted) [reply docs/DeveloperGuide.md]: This issue has been resolved. Please do check again :)

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (Berted) [reply docs/DeveloperGuide.md]: This issue has been resolved. Please do check again :)

**Review by optionalemon — APPROVED** [review]
LGTM!

**Pull request merged**